CapstoneProject/Assets/Scripts/Core/Puzzle/TimedDoor.cs

```diff
@@ -0,0 +1,85 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.UI;
+using TMPro;
+using System;
+
+public class TimedDoor : MonoBehaviour
```

Comment:
you should add a requires component of collidee2d to this script if we assume it has one in functions below